app/controllers/subscriptions_controller.rb

```diff
@@ -23,14 +27,14 @@ def create
           source: existing_subscription ? :frequency_changed : :user_signed_up,
         )
       end
+    end
 
-      subscription = new_subscription || existing_subscription
-      email = if create_new_subscription
-                SubscriptionConfirmationEmailBuilder.call(subscription: subscription)
-              end
+    subscription = new_subscription || existing_subscription
+    email = if create_new_subscription
```

Comment:
I think we can do this in the transaction too: it's all inside the system. We take a similar approach [here](https://github.com/alphagov/email-alert-api/pull/1463/files#diff-547e678b8e52e4f0db33fa3d82cf46ed9ebbcca4aad50aefc889ce4ef2b168a8L33). If this becomes more involved, we could also separate this into a different PR, as it's tangential to the overall change.

Reply:
I wasn't bothered about adding the email generation bits inside the transaction as we will be removing the use of this soon [here](https://trello.com/c/DkVHm6Gs/621-remove-subscription-confirmation-email). So I think is fine as is?

Reply:
Okay sure, have added it back! 👍